### DISCUSSION TITLE
How to efficiently process a batch of terms in one go for Tree Neural Networks?

### DISCUSSION THREAD

**Discussion by brando90** [princeton-vl/CoqGym#48] [Q&A]
this seems like an important point so I wanted to make a discussion. I was wondering if the code that batches ASTs has really been tested to see if it's faster?

https://github.com/princeton-vl/CoqGym/issues/16

**Comment by yangky11** [comment]
I may have measured the speed and found it to be faster when I wrote the code. I'm not sure though at this point.

**Reply by brando90** [reply]
I am curious, is there a reason you didn't use these two options for batching:

- https://developer.nvidia.com/blog/recursive-neural-networks-pytorch/
- https://github.com/unbounce/pytorch-tree-lstm

?

**Reply by yangky11** [reply]
I was unaware of these two repos. Also, since batching for TreeLSTM is not too complex, it was not worth introducing another 3rd-party dependency.
